### ISSUE TITLE
Simpler print output?

### ISSUE THREAD

**Issue by jgillis** [casadi/casadi#1576]
````
@1=1, 
[[(@1-sq(q)), (-1-(p*(q+q))), 00], 
 [@1, 00, 00], 
 [(p+p), (q+q), 00]]
```

**Comment by jgillis** [comment]
Related, the printDimensions output is crowded to read.
This reads much friendlier:
```
***********f************ || ************ffwd************ || ************fadj************
 Number of inputs: 3     ||  Number of inputs: 7         ||  Number of inputs: 5        
  0. i0, 1-by-1 (dense)  ||   0. der_i0, 1-by-1 (dense)  ||   0. der_i0, 1-by-1 (dense) 
  1. i1, 1-by-1 (dense)  ||   1. der_i1, 1-by-1 (dense)  ||   1. der_i1, 1-by-1 (dense) 
  2. i2, 3-by-1 (dense)  ||   2. der_i2, 3-by-1 (dense)  ||   2. der_i2, 3-by-1 (dense) 
 Number of outputs: 1    ||   3. der_o0, 3-by-1 (0/3 nz) ||   3. der_o0, 3-by-1 (0/3 nz)
  0. o0, 3-by-1 (dense)  ||   4. fwd0_i0, 1-by-1 (dense) ||   4. adj0_o0, 3-by-1 (dense)
@0 = input[2][1];        ||   5. fwd0_i1, 1-by-1 (dense) ||  Number of outputs: 3       
@1 = sq(@0);             ||   6. fwd0_i2, 3-by-1 (dense) ||   0. adj0_i0, 1-by-1 (dense)
@2 = 1;                  ||  Number of outputs: 1        ||   1. adj0_i1, 1-by-1 (dense)
@2 = (@2-@1);            ||   0. fwd0_o0, 3-by-1 (dense) ||   2. adj0_i2, 3-by-1 (dense)
```

**Comment by jaeandersson** [comment]
I think it's good. It's very compact and descriptive.

**Comment by jaeandersson** [comment]
Note that the above printout is much more compact than printing the algorithm.

**Comment by jgillis** [comment]
Let me clarify, I am discussing only subexpressions that are a pure constant. I'm not critizing the new style of printing. I feel like "@1=1" does not add value to the printout since constants are always cached anyway. They substract value by making it appear more cluttered, imho.

Constrast with:
```
[[(1-sq(q)), (-1-(p*(q+q))), 00], 
 [1, 00, 00], 
 [(p+p), (q+q), 00]]
```

**Comment by jaeandersson** [comment]
Now replace 1 with 1.000001 and you get:
````
@1=1.000001, 
[[(@1-sq(q)), (-1-(p*(q+q))), 00], 
 [@1, 00, 00], 
 [(p+p), (q+q), 00]]
```
vs.
```` 
[[(1.000001-sq(q)), (-1-(p*(q+q))), 00], 
 [1.000001, 00, 00], 
 [(p+p), (q+q), 00]]
```

There are also more subtile differences, for example if two constants are *almost* equal, say 1.000001 and 1.000002. They might have the same printout, but with the `@` syntax you can notice that they are different (otherwise they would point to the same subexpression due to caching).

Another issue is the fact that constants are cached in SX, but not in MX. So in MX, they might actually be different.

In short, I think the current syntax is good as it is.
